lib/src/main/kotlin/com/swmansion/starknet/service/http/OkhttpHttpService.kt

```diff
@@ -12,9 +12,13 @@ import java.io.IOException
 import java.util.concurrent.CompletableFuture
 
 /**
- * Service for making http requests.
+ * Service for making http requests using OkHttp library. You can reuse provide it with your client to
+ * <a href="https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/#okhttpclients-should-be-shared"> avoid
+ * wasting resources</a>.
+ * @param client OkHttpClient used for making requests
  */
-internal class OkhttpHttpService() : HttpService {
+class OkhttpHttpService(private val client: OkHttpClient) : HttpService {
```

Comment:
Users can now provide their own `OkHttpClient` in here and pass it to providers.